Patch to prevent FTBFS on AMD64 because of -Werror.  Based on upstream
commits 3b93dcc4726d4363215233a780df34772d76ee09 and
6555359f8e56b6e08bd1a8f40fd363506a0e3ff5

diff --git a/src/network/ssl/socket.c b/src/network/ssl/socket.c
index 19b8941..cef1164 100644
--- a/src/network/ssl/socket.c
+++ b/src/network/ssl/socket.c
@@ -155,7 +155,7 @@ ssl_connect(struct socket *socket)
 
 #elif defined(CONFIG_GNUTLS)
 	gnutls_transport_set_ptr(*((ssl_t *) socket->ssl),
-				 (gnutls_transport_ptr) socket->fd);
+				 (gnutls_transport_ptr) (longptr_T) socket->fd);
 
 	/* TODO: Some certificates fuss. --pasky */
 #endif
diff --git a/src/osdep/generic.h b/src/osdep/generic.h
index 9223cd5..357b602 100644
--- a/src/osdep/generic.h
+++ b/src/osdep/generic.h
@@ -102,7 +102,7 @@ safe_write(int fd, const void *buf, size_t count) {
 
 /* Alignment of types.  */
 #define alignof(TYPE) \
-    ((int) &((struct { unsigned char dummy1; TYPE dummy2; } *) 0)->dummy2)
+    offsetof(struct { unsigned char dummy1; TYPE dummy2; }, dummy2)
 
 /* Using this macro to copy structs is both faster and safer than
  * memcpy(destination, source, sizeof(source)). Please, use this macro instead
